Document how to switch to a C++ Project

It would be helpful if it were documented what needed to be done to switch to a C++ program.

From what I've read, renaming `main.c` to `main.cpp` is important, but beyond that I'm unsure what additional CMake changes need to be made to get everything (especially the STM32 HAL) to play nicely together.

For example, after making this change I seem to get linker issues building for the STM32F3DISCOVERY:

```
...
CMakeFiles/Project.elf.dir/startup/startup_stm32f303xc.s.obj: In function `LoopFillZerobss':
Project/startup/startup_stm32f303xc.s:111: undefined reference to `main'
```

--- a/resources/xyz/elmot/clion/cubemx/tmpl_CMakeLists.txt
+++ b/resources/xyz/elmot/clion/cubemx/tmpl_CMakeLists.txt
@@ -29,6 +29,7 @@ SET(CMAKE_CXX_FLAGS_INIT "$${COMMON_FLAGS} -std=c++11")
 SET(CMAKE_C_FLAGS_INIT "$${COMMON_FLAGS} -std=gnu99")
 SET(CMAKE_EXE_LINKER_FLAGS_INIT "-Wl,-gc-sections,--print-memory-usage -T $${LINKER_SCRIPT}")
 
+# To switch to a C++ project, add CXX to the PROJECT language list command and rename main.c to main.cpp.
 PROJECT(${projectName} C ASM)
 set(CMAKE_CXX_STANDARD 11)
 

--- a/resources/xyz/elmot/clion/cubemx/tmpl_CMakeLists.txt
+++ b/resources/xyz/elmot/clion/cubemx/tmpl_CMakeLists.txt
@@ -29,8 +29,7 @@ SET(CMAKE_CXX_FLAGS_INIT "$${COMMON_FLAGS} -std=c++11")
 SET(CMAKE_C_FLAGS_INIT "$${COMMON_FLAGS} -std=gnu99")
 SET(CMAKE_EXE_LINKER_FLAGS_INIT "-Wl,-gc-sections,--print-memory-usage -T $${LINKER_SCRIPT}")
 
-# To switch to a C++ project, add CXX to the PROJECT language list command and rename main.c to main.cpp.
-PROJECT(${projectName} C ASM)
+PROJECT(${projectName} C CXX ASM)
 set(CMAKE_CXX_STANDARD 11)
 
 #add_definitions(-DARM_MATH_CM4 -DARM_MATH_MATRIX_CHECK -DARM_MATH_ROUNDING -D__FPU_PRESENT=1 )

--- a/resources/xyz/elmot/clion/cubemx/tmpl_CMakeLists.txt
+++ b/resources/xyz/elmot/clion/cubemx/tmpl_CMakeLists.txt
@@ -6,6 +6,7 @@ cmake_minimum_required(VERSION 3.7)
 # specify the cross compiler
 SET(CMAKE_C_COMPILER_WORKS 1)
 SET(CMAKE_C_COMPILER arm-none-eabi-gcc)
+SET(CMAKE_CXX_COMPILER_WORKS 1)
 SET(CMAKE_CXX_COMPILER arm-none-eabi-g++)
 set(AS arm-none-eabi-as)
 set(AR arm-none-eabi-ar)

